tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1046,8 +1054,11 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
   Status FuseMatrixBiasAdd(HloInstruction *instr, HloInstruction *bias,
                            const HloInstruction *gemm,
-                           HloInstruction *bitcast = nullptr) {
-    TF_RET_CHECK(bias->shape() == (bitcast ? bitcast->shape() : gemm->shape()));
+                           HloInstruction *bitcast = nullptr,
+                           HloInstruction *slice = nullptr) {
+    TF_RET_CHECK(
+        (bias->shape() == (bitcast ? bitcast->shape() : gemm->shape())) ||
+        (bias->shape() == (slice ? slice->shape() : gemm->shape())));
```

Comment:
Only check the slice shape if there is no bitcast, since otherwise this condition is confusing. You can write this as:

```c++
TF_RET_CHECK(bias->shape() == (bitcast ? bitcast->shape() : slice ? slice->shape() : gemm->shape()));
```

Reply:
Resolved.